docs/best-practices.md

```diff
@@ -0,0 +1,82 @@
+    - Open parameters, only if they are required to open the event source
+- Follow the secret management guidelines imposed by well-known packages such as [aws-sdk-go](https://github.com/aws/aws-sdk-go#configuring-credentials)
+
+
+## Managing the NextBatch Loop
+
+In source plugins, new events are created and returned in batches with the `plugin_next_batch()` function. This design decision reduces the overhead that may occur by integrating C with other programming languages such as Go. Although the SDKs cover most of the hard work, plugin developers must manage the batching system manually and there are some things to keep in mind in order to ensure optimal performance and correct behavior.
+
+### Don't
+
+- Allocate a new event batch at every invokation of `plugin_next_batch()`, because event generation is a very hot path
+- Block the execution for long times: be mindful that `plugin_next_batch()` is blocking for the framework
+- Return an EOF error and then return new events at the following invokation of `plugin_next_batch()`
+- Return a given number of events without filling their data in the event batch
+
+### Do
+
+- Rely on the Go SDK and C++ SDK to ensure that all memory allocations are optimized
+- Return a number of events smaller than the size of the batch, if fastly available (see the behavior described [in the Go SDK](https://pkg.go.dev/github.com/falcosecurity/plugin-sdk-go@v0.1.0/pkg/sdk#NextBatcher))
+- Return Timeout (see [`sdk.ErrTimeout`](https://github.com/falcosecurity/plugin-sdk-go/blob/0b4b6dc215141116c53398f3232aac98e49cdb80/pkg/sdk/sdk.go#L30) in Go or [`SS_PLUGIN_TIMEOUT`](https://github.com/falcosecurity/libs/blob/033c4b9f28e58e20a5822bd8a7419beea098af91/userspace/libscap/plugin_info.h#L76) in C/C++) after a short time period passes without producing any event (no more than few milliseconds)
+- Return Timeout even with a non-zero number of events, if they are available
+- Return EOF (see [`sdk.ErrEOF`](https://github.com/falcosecurity/plugin-sdk-go/blob/0b4b6dc215141116c53398f3232aac98e49cdb80/pkg/sdk/sdk.go#L25) in Go or [`SS_PLUGIN_EOF`](https://github.com/falcosecurity/libs/blob/033c4b9f28e58e20a5822bd8a7419beea098af91/userspace/libscap/plugin_info.h#L77) in C/C++) when you are certain that no more events will be produced
+- Return EOF with a non-zero number of events, but ensuring that every following invokation of `plugin_next_batch()` returns EOF and zero events
+
+
+## Fields Naming Conventions
+
+Plugins can extend Falco by supporting new fields to be extracted event data. At runtime, those fields co-exist with the other fields natively supported in libsinsp, such as the `evt.*` or the `container.*` classes. In this perspective, it's really important for plugin developers to follow some naming conventions for the newly introduced fields.
+
+### Don't
+
+- Introduce fields with the same name or the same class (e.g. `evt.*`, `container.*`, `ka.*`) as the ones already existing in libsinsp
```

Comment:
This may contrast with https://falco.org/docs/plugins/#handling-duplicate-overlapping-fields-in-plugins-libraries-core

Reply:
In my opinion it doesn't. The docs at falco.org are more tied to technical details, whereas this section is about best practices. In other words, this line does not imply that developers **can't** introduce overlapping filterchecks (which is technically feasible for different sources), but just state that it's better to not create ambiguity by introducing new fields with the same names as the ones of libhsinsp.